src/nvim/api/win_config.c

```diff
@@ -222,9 +223,11 @@ void nvim_win_set_config(Window window, Dict(float_config) *config, Error *err)
     win_config_float(win, fconfig);
     win->w_pos_changed = true;
   }
-  if (fconfig.style == kWinStyleMinimal) {
-    win_set_minimal_style(win);
-    didset_window_options(win, true);
+  if (prev_style != fconfig.style) {
```

Comment:
I think if you explicitly pass-in `style="minimal"` a second time it should still apply again.

Reply:
Updated.